gravitee-apim-gateway/gravitee-apim-gateway-tcp/src/main/java/io/gravitee/gateway/reactive/tcp/VertxTcpClient.java

```diff
@@ -0,0 +1,84 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.gateway.reactive.tcp;
+
+import io.gravitee.definition.model.v4.ssl.SslOptions;
+import io.gravitee.definition.model.v4.ssl.jks.JKSKeyStore;
+import io.gravitee.definition.model.v4.ssl.jks.JKSTrustStore;
+import io.gravitee.definition.model.v4.ssl.pem.PEMKeyStore;
+import io.gravitee.definition.model.v4.ssl.pem.PEMTrustStore;
+import io.gravitee.definition.model.v4.ssl.pkcs12.PKCS12KeyStore;
+import io.gravitee.definition.model.v4.ssl.pkcs12.PKCS12TrustStore;
+import io.gravitee.definition.model.v4.tcp.TcpClientOptions;
+import io.gravitee.definition.model.v4.tcp.TcpTarget;
+import io.vertx.core.net.JksOptions;
+import io.vertx.core.net.NetClientOptions;
+import io.vertx.core.net.PemKeyCertOptions;
+import io.vertx.core.net.PemTrustOptions;
+import io.vertx.core.net.PfxOptions;
+import io.vertx.rxjava3.core.Vertx;
+import io.vertx.rxjava3.core.net.NetClient;
+import java.util.Base64;
+import lombok.Builder;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@Builder
+public class VertxTcpClient extends AbstractBaseClient {
+
+    @NonNull
+    private final Vertx vertx;
+
+    private final TcpTarget tcpTarget;
+    private TcpClientOptions tcpOptions;
+    private SslOptions sslOptions;
+
+    public NetClient createTcpClient() {
+        if (tcpOptions == null) {
+            tcpOptions = new TcpClientOptions();
+        }
+
+        return vertx.createNetClient(createNetClientOptions());
+    }
+
+    private NetClientOptions createNetClientOptions() {
+        var clientOptions = new NetClientOptions().setMetricsName("tcp-client");
+        clientOptions
+            .setConnectTimeout(tcpOptions.getConnectTimeout())
+            .setReconnectAttempts(tcpOptions.getReconnectAttempts())
+            .setReconnectInterval(tcpOptions.getReconnectInterval());
+
+        configureSsl(clientOptions);
+
+        return clientOptions;
+    }
+
+    private void configureSsl(final NetClientOptions clientOptions) {
+        if (sslOptions != null) {
+            clientOptions.setSsl(true);
+            clientOptions.setTrustAll(sslOptions.isTrustAll());
+
+            if (!sslOptions.isTrustAll()) {
+                // Client truststore configuration (trust server certificate).
+                super.configureTrustStore(clientOptions, sslOptions, tcpTarget.toString());
+
+                // Client keystore configuration (client certificate for mtls).
+                super.configureKeyStore(clientOptions, sslOptions, tcpTarget.toString());
```

Comment:
Should it be outside of the `if` instead?

Reply:
I did it as it's done in VertxHttpClient, do you think it could be something to fix also on this class ?

Reply:
Not the same condition

`if (!sslOptions.isTrustAll()) {`

versus

`if (sslOptions != null) {`